test/e2e.js

```diff
@@ -0,0 +1,100 @@
+'use strict'
+'use strict'
+
+const test = require('node:test')
+const assert = require('node:assert')
+const { readFile } = require('node:fs/promises')
+const { join } = require('node:path')
+
+const fastify = require('fastify')
+const mtlsPlugin = require('..')
+const { MockAgent, setGlobalDispatcher, request } = require('undici')
+const UndiciTLSDispatcher = require('undici-tls-dispatcher')
+
+const fixturesDirectory = join(__dirname, 'fixtures', 'e2e')
+
+test('should implement mTLS', async (t) => {
+  const { url, server } = await buildServer()
+  const key = await readFile(join(fixturesDirectory, 'client', 'key.pem'), 'utf-8')
+  const ca = await readFile(join(fixturesDirectory, 'client', 'ca.pem'), 'utf-8')
+  const cert = await readFile(join(fixturesDirectory, 'client', 'cert.pem'), 'utf-8')
+  const dispatcher = new UndiciTLSDispatcher({
+    tlsConfig: [
+      {
+        url,
+        tls: {
+          ca,
+          cert,
+          key
+        }
+      }
+    ]
+  })
+  const res = await request(`${url}/mtls`, { dispatcher })
+  const body = await res.body.json()
+  assert.strictEqual(body.message, 'You are client.test.fly.dev') // fixtures certificate is valid for this host
+
+  // should not connect without certificate
+  try {
+    await request(`${url}/mtls`)
+  } catch (err) {
+    assert.strictEqual(err.code, 'SELF_SIGNED_CERT_IN_CHAIN')
```

Comment:
This message is correct, because the server presents a certificate signed by a CA, which in turn has a self-signed certificate. Note that every root CA has a  self-signed certificate, the difference is if the client has the certificate's CA (which has not, in this case)